### PR TITLE
Record and Display Data on Missing Files

### PR DESCRIPTION
**Pull request by Dala0**
## What does this change?

Adds recording and display of data on files found in the project that are not in the asset folder.

## How can we measure success?

Data is recorded on the missing files. The recorded data is displayed in the interface.

## Images

PC115

This was tested on a machine running macOS 12.6.8 under minikube 1.31.2 and on the dev. system.

### CODE REVIEW

File: app/services/MissingFilesReceiver.scala [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                channel.basicAck(envelope.getDeliveryTag, false)
+              case false=>
+                logger.warn("Could not handle message, leaving it on-queue")
+                channel.basicNack(envelope.getDeliveryTag, false, true)
+            })
+        }
+      }
+
+      def handleEvent(messageData: MissingFile): Future[Boolean] = {
+        if (messageData.project_id.toInt.isValidInt) {
```
Comment (richpryce) [inline app/services/MissingFilesReceiver.scala:75]: You could avoid the repetitive `toInt` conversions by doing something like `val projectId = messageData.project_id.toInt`

`if (!projectId.isValidInt) Future.successful(false)`

File: app/services/MissingFilesReceiver.scala [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+          if (messageData.file_path.startsWith("/")) {
+            if (messageData.file_path.startsWith("/Volumes/Multimedia2/Media Production/Assets/Branding")) {
```
Comment (richpryce) [inline app/services/MissingFilesReceiver.scala:77]: You could refactor the nested conditionals into separate functions, making the main function more readable

**Review by richpryce — COMMENTED** [review]
Looks ok but could do with some unit / integration tests

**Comment by Dala0** [comment]
> Looks ok but could do with some unit / integration tests

What do you want tested?

**Pull request merged**